src/lib/private_components/layered-map-resources/DrawControls.react.js

```diff
@@ -0,0 +1,120 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { withLeaflet } from 'react-leaflet';
+import  {EditControl}  from 'react-leaflet-draw';
+import "./assets/leaflet.draw.css";
+import L from 'leaflet';
+
+
+// work around broken icons when using webpack, see https://github.com/PaulLeCam/react-leaflet/issues/255
+
+delete L.Icon.Default.prototype._getIconUrl;
+L.Icon.Default.mergeOptions({
+  iconRetinaUrl: require('leaflet/dist/images/marker-icon-2x.png'),
+  iconUrl: require('leaflet/dist/images/marker-icon.png'),
+  shadowUrl: require('leaflet/dist/images/marker-shadow.png'),
+})
+
+/**  Leaflet-draw:edit event does not return marker type.
+*    Helper function to find marker type.
+*    https://stackoverflow.com/questions/18014907/leaflet-draw-retrieve-layer-type-on-drawedited-event
+**/
+var getShapeType = (layer) => {
+
+    if (layer instanceof L.Circle) {
+        return 'circle';
+    }
+
+    if (layer instanceof L.Marker) {
+        return 'marker';
+    }
+
+    if ((layer instanceof L.Polyline) && ! (layer instanceof L.Polygon)) {
+        return 'polyline';
+    }
+
+    if ((layer instanceof L.Polygon) && ! (layer instanceof L.Rectangle)) {
+        return 'polygon';
+    }
+
+    if (layer instanceof L.Rectangle) {
+        return 'rectangle';
+    }
+    return null
```

Comment:
This whole function could perhaps be replaced by a switch statement? If at the same time reodered such that special cases (e.g. rectangle) are checked before general instance (e.g. polygon).